### PR TITLE
Added PreValidate endpoint on Account controller

### PR DESCRIPTION
**Pull request by cscharf**
## Overview
Added a known endpoint to the Identity service for validating a domain hint/org identifier for SSO. This endpoint basically calls and tests against the SSO service's endpoint of the same name and path and curries the value back to the caller. This is done before the Identity service endpoint/IURL is known to the Web and other clients, however the SSO endpoint is not, it is an intermediary SP between Identity service and an external IdP for SSO _and_ the SSO service is the one that actually loads and validates and contains the appropriate handlers and authentication schemes to validate against at a deeper level.

## Changes
* Cleanup of repository for Organizations and sproc
* New pre-validation endpoint for the Account controller
* Added service specific injector of an `HttpClient` to the `AccountController`

### CODE REVIEW

File: util/Migrator/DbScripts/2020-08-12_00_OrgIdentifierProc.sql [hunk 1]
```diff
@@ -5,7 +5,7 @@ END
 GO
 
 CREATE PROCEDURE [dbo].[Organization_ReadByIdentifier]
-    @Identifier UNIQUEIDENTIFIER
+    @Identifier NVARCHAR(50)
```
Comment (kspearrin) [inline util/Migrator/DbScripts/2020-08-12_00_OrgIdentifierProc.sql:8]: We should correct this with a new script at this point.

File: src/Identity/Controllers/AccountController.cs [hunk 2]
```diff
@@ -26,21 +28,61 @@ public class AccountController : Controller
...
+            {
+                Response.StatusCode = 400;
+                return Json(new ErrorResponseModel("No domain hint was provided"));
+            }
+            try
+            {
+                // Calls Sso Pre-Validate, assumes baseUri set
+                var requestPath = $"/Account/PreValidate?domainHint={domainHint}";
+                var httpClient = _clientFactory.CreateClient("InternalSso");
+                using var responseMessage = await httpClient.GetAsync(requestPath);
```
Comment (kspearrin) [inline src/Identity/Controllers/AccountController.cs:65]: I see you added i18n to the sso service. Won't you need to forward the language header coming from the client?

File: util/Migrator/DbScripts/2020-08-12_00_OrgIdentifierProc.sql (hunk 1 above)
Comment (cscharf) [reply util/Migrator/DbScripts/2020-08-12_00_OrgIdentifierProc.sql:8]: Done

File: src/Identity/Controllers/AccountController.cs (hunk 2 above)
Comment (cscharf) [reply src/Identity/Controllers/AccountController.cs:68]: Done (now)

**Comment by cscharf** [comment]
ahh, sorry, found some bugs/issues and made some updates accordingly to finish up, my bad.

**Pull request merged**